AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientExtensions.swift

```diff
@@ -745,15 +745,16 @@ extension AWSMobileClient {
             self.tokenFetchOperationQueue.addOperation {
                 self.tokenFetchLock.enter()
                 AWSCognitoAuth.init(forKey: self.CognitoAuthRegistrationKey).getSession({ (session, error) in
-                    if let error = error as? AWSCognitoAuthClientErrorType {
-                        if error == AWSCognitoAuthClientErrorType.errorExpiredRefreshToken {
-                            self.pendingGetTokensCompletion = completionHandler
-                            self.mobileClientStatusChanged(userState: .signedOutUserPoolsTokenInvalid, additionalInfo: [self.ProviderKey:"OAuth"])
-                            // return early without releasing the tokenFetch lock.
-                            return
-                        } else {
-                            completionHandler(nil, AWSMobileClientError.makeMobileClientError(from: error))
-                        }
+
+                    if let sessionError = error,
+                        (sessionError as NSError).domain == AWSCognitoAuthErrorDomain,
+                        let errorType = AWSCognitoAuthClientErrorType(rawValue: (sessionError as NSError).code),
+                        (errorType == .errorExpiredRefreshToken) {
```

Comment:
maybe `if isExpiredRefreshToken(error)` if it can be reused in other places?

Reply:
After this change HostedUI and non-HostedUI should work the same. ie Whenever the session expires they will be notified by the user listener.